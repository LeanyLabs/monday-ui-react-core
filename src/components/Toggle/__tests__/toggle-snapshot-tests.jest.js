import React from "react";
import renderer from "react-test-renderer";
import Toggle from "../Toggle";


describe("Toggle renders correctly", () => {
  it("renders correctly with empty props", () => {
    const tree = renderer.create(<Toggle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when selection defined by default selected (on) ", () => {
    const tree = renderer.create(<Toggle componentClassName="dummy-class-name" isDefaultSelected />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when selection defined by default selected (off) ", () => {
    const tree = renderer.create(<Toggle componentClassName="dummy-class-name" isDefaultSelected={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when selection defined by isSelected (on)", () => {
    const tree = renderer.create(<Toggle componentClassName="dummy-class-name" isSelected />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when selection defined by isSelected (off)", () => {
    const tree = renderer.create(<Toggle componentClassName="dummy-class-name" isSelected={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when disabled", () => {
    const tree = renderer.create(<Toggle isDisabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
