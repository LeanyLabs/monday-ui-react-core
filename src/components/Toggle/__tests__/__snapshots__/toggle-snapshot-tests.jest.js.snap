// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders correctly renders correctly when disabled 1`] = `
<label
  className="monday-style-toggle_wrapper monday-style-toggle_wrapper--disabled"
>
  <span
    className="monday-style-toggle_text"
  >
    Off
  </span>
  <input
    aria-checked={true}
    checked={true}
    className="monday-style-toggle_input"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="switch"
    type="checkbox"
  />
  <div
    aria-hidden="true"
    className="monday-style-toggle_toggle monday-style-toggle_toggle--selected"
  />
  <span
    className="monday-style-toggle_text"
  >
    On
  </span>
</label>
`;

exports[`Toggle renders correctly renders correctly when labels are hidden 1`] = `
<label
  className="monday-style-toggle_wrapper"
>
  <input
    aria-checked={true}
    checked={true}
    className="monday-style-toggle_input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="switch"
    type="checkbox"
  />
  <div
    aria-hidden="true"
    className="monday-style-toggle_toggle monday-style-toggle_toggle--selected"
  />
</label>
`;

exports[`Toggle renders correctly renders correctly when selection defined by default selected (off)  1`] = `
<label
  className="monday-style-toggle_wrapper"
>
  <span
    className="monday-style-toggle_text"
  >
    Off
  </span>
  <input
    aria-checked={false}
    checked={false}
    className="monday-style-toggle_input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="switch"
    type="checkbox"
  />
  <div
    aria-hidden="true"
    className="monday-style-toggle_toggle dummy-class-name monday-style-toggle_toggle--not-selected"
  />
  <span
    className="monday-style-toggle_text"
  >
    On
  </span>
</label>
`;

exports[`Toggle renders correctly renders correctly when selection defined by default selected (on)  1`] = `
<label
  className="monday-style-toggle_wrapper"
>
  <span
    className="monday-style-toggle_text"
  >
    Off
  </span>
  <input
    aria-checked={true}
    checked={true}
    className="monday-style-toggle_input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="switch"
    type="checkbox"
  />
  <div
    aria-hidden="true"
    className="monday-style-toggle_toggle dummy-class-name monday-style-toggle_toggle--selected"
  />
  <span
    className="monday-style-toggle_text"
  >
    On
  </span>
</label>
`;

exports[`Toggle renders correctly renders correctly when selection defined by isSelected (off) 1`] = `
<label
  className="monday-style-toggle_wrapper"
>
  <span
    className="monday-style-toggle_text"
  >
    Off
  </span>
  <input
    aria-checked={false}
    checked={false}
    className="monday-style-toggle_input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="switch"
    type="checkbox"
  />
  <div
    aria-hidden="true"
    className="monday-style-toggle_toggle dummy-class-name monday-style-toggle_toggle--not-selected"
  />
  <span
    className="monday-style-toggle_text"
  >
    On
  </span>
</label>
`;

exports[`Toggle renders correctly renders correctly when selection defined by isSelected (on) 1`] = `
<label
  className="monday-style-toggle_wrapper"
>
  <span
    className="monday-style-toggle_text"
  >
    Off
  </span>
  <input
    aria-checked={true}
    checked={true}
    className="monday-style-toggle_input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="switch"
    type="checkbox"
  />
  <div
    aria-hidden="true"
    className="monday-style-toggle_toggle dummy-class-name monday-style-toggle_toggle--selected"
  />
  <span
    className="monday-style-toggle_text"
  >
    On
  </span>
</label>
`;

exports[`Toggle renders correctly renders correctly with aria controls 1`] = `
<label
  className="monday-style-toggle_wrapper"
>
  <span
    className="monday-style-toggle_text"
  >
    Off
  </span>
  <input
    aria-checked={true}
    aria-controls="aria controls"
    checked={true}
    className="monday-style-toggle_input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="switch"
    type="checkbox"
  />
  <div
    aria-hidden="true"
    className="monday-style-toggle_toggle monday-style-toggle_toggle--selected"
  />
  <span
    className="monday-style-toggle_text"
  >
    On
  </span>
</label>
`;

exports[`Toggle renders correctly renders correctly with empty props 1`] = `
<label
  className="monday-style-toggle_wrapper"
>
  <span
    className="monday-style-toggle_text"
  >
    Off
  </span>
  <input
    aria-checked={true}
    checked={true}
    className="monday-style-toggle_input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="switch"
    type="checkbox"
  />
  <div
    aria-hidden="true"
    className="monday-style-toggle_toggle monday-style-toggle_toggle--selected"
  />
  <span
    className="monday-style-toggle_text"
  >
    On
  </span>
</label>
`;

exports[`Toggle renders correctly renders correctly with not default offOverrideText 1`] = `
<label
  className="monday-style-toggle_wrapper"
>
  <span
    className="monday-style-toggle_text"
  >
    TestOff
  </span>
  <input
    aria-checked={true}
    checked={true}
    className="monday-style-toggle_input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="switch"
    type="checkbox"
  />
  <div
    aria-hidden="true"
    className="monday-style-toggle_toggle monday-style-toggle_toggle--selected"
  />
  <span
    className="monday-style-toggle_text"
  >
    On
  </span>
</label>
`;

exports[`Toggle renders correctly renders correctly with not default onOverrideText 1`] = `
<label
  className="monday-style-toggle_wrapper"
>
  <span
    className="monday-style-toggle_text"
  >
    Off
  </span>
  <input
    aria-checked={true}
    checked={true}
    className="monday-style-toggle_input"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="switch"
    type="checkbox"
  />
  <div
    aria-hidden="true"
    className="monday-style-toggle_toggle monday-style-toggle_toggle--selected"
  />
  <span
    className="monday-style-toggle_text"
  >
    TestOn
  </span>
</label>
`;
